test/gateway/safe_core/set_co_core_address.js

```diff
@@ -0,0 +1,118 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const SafeCore = artifacts.require("./SafeCore.sol");
+const MockMembersManager = artifacts.require('MockMembersManager.sol');
+const web3 = require('../../test_lib/web3.js');
+const BN = require('bn.js');
+const Utils = require('../../../test/test_lib/utils');
+
+const NullAddress = "0x0000000000000000000000000000000000000000";
+
+contract('SafeCore.setCoCoreAddress()', function (accounts) {
+
+  let remoteChainId,
+    blockHeight,
+    stateRoot,
+    membersManager,
+    safeCore,
+    owner,
+    worker,
+    coCoreAddress;
+
+  beforeEach(async function () {
+
+    owner = accounts[2];
+    worker = accounts[3];
+    remoteChainId = new BN(1410);
+    blockHeight = new BN(5);
+    stateRoot = web3.utils.sha3("dummy_state_root");
+    membersManager = await MockMembersManager.new(owner, worker);
+    coCoreAddress = accounts[6];
+
+    safeCore = await SafeCore.new(
+      remoteChainId,
+      blockHeight,
+      stateRoot,
+      membersManager.address,
+    );
+
+  });
+
+
+  it('should fail to set coCore address zero', async () => {
```

Comment:
should fail to set coCore address zero ==> should fail when coCore address is set to zero